### PR TITLE
Feature led matrix

### PR DESCRIPTION
**Pull request by Mikscho**

### CODE REVIEW

**Comment by RetoStutz** [comment]
Webpage:
Where is the page for the led matrix?
reference the led matrix page in the led strip page

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        this(pi4j, new int[Rows][Columns], brightness, DEFAULT_CHANNEL);
+    }
+
+    /**
+     * Creates a new simpleLed component with a custom BCM pin.
+     *
+     * @param pi4j    Pi4J context
+     * @param matrix How many LEDs are on this Strand
+     * @param brightness How bright the leds can be at max, Range 0 - 255
+     * @param channel which channel to use
```
Comment (RetoStutz) [inline src/main/java/com/pi4j/example/components/LEDMatrix.java:57]: what is the meaning of channel? would it be better to name it spi channel?

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        for (int[] ints : matrix) {
+            this.numLeds += ints.length;
+        }
+
+        this.ledStrip = new LEDStrip(pi4j, numLeds, brightness, channel);
+        this.spi = ledStrip.spi;
+    }
+
+    /**
+     * @return the pi4j context
```
Comment (RetoStutz) [inline src/main/java/com/pi4j/example/components/LEDMatrix.java:75]: PI4J

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+     */
+    public Context getContext() {
+        return this.context;
+    }
+
+    /**
+     * Setting all LEDS off and closing the strip
```
Comment (RetoStutz) [inline src/main/java/com/pi4j/example/components/LEDMatrix.java:82]: leds or LEDs

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        Arrays.fill(matrix[strip], color);
+    }
+
+    /**
+     * Setting all leds to the same color
+     *
+     * @param color the color that is set
+     */
+    public void setMatrixColor(int color) {
+        for (int[] ints : matrix) {
```
Comment (RetoStutz) [inline src/main/java/com/pi4j/example/components/LEDMatrix.java:153]: what is ints? LEDs?

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+     */
+    public void allOff() {
+        ledStrip.allOff();
+    }
+
+    /**
+     * Utility function to sleep for the specified amount of milliseconds. An {@link InterruptedException} will be catched and ignored while setting the
+     * interrupt flag again.
+     */
+    protected void sleep(long millis, int nanos) {
```
Comment (RetoStutz) [inline src/main/java/com/pi4j/example/components/LEDMatrix.java:183]: why do you need this function?

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+     * @param pixel which position on the strip, range 0 - numLEDS-1
+     * @param color the color that is set
+     */
+    public void setPixelColor(int strip, int pixel, int color) {
+        if (strip > matrix.length || strip < 0 || pixel > matrix[strip].length || pixel < 0) {
+            throw new IllegalArgumentException("the strip or led specified does not exist");
+        }
+        matrix[strip][pixel] = color;
+    }
+
```
Comment (RetoStutz) [inline src/main/java/com/pi4j/example/components/LEDMatrix.java:134]: new function set/get PixelColor(int pixelNumber, int color);

to address a led by its number in the led strip

Comment (Mikscho) [reply src/main/java/com/pi4j/example/components/LEDMatrix.java:134]: created

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+     * You can give in something like int[3][4] or
+     * matrix = {{0, 0, 0},
+     * {0, 0, 0, 0},
+     * {0}}
+     *
+     * @param pi4j    Pi4J context
+     * @param matrix How many LEDs are on this Strand
+     * @param brightness How bright the leds can be at max, Range 0 - 1
+     */
+    public LEDMatrix(Context pi4j, int[][] matrix, double brightness) {
```
[the remark this answers is not in the document]
Comment (Mikscho) [reply src/main/java/com/pi4j/example/components/LEDMatrix.java:37]: nope, you will control the whole strip with brightness. you can control the brightness by choosing another color-int-value

File: src/main/java/com/pi4j/example/components/LEDMatrix.java [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+     * @param brightness How bright the leds can be at max, Range 0 - 1
+     */
+    public LEDMatrix(Context pi4j, int Rows, int Columns, double brightness) {
+        this(pi4j, new int[Rows][Columns], brightness, DEFAULT_CHANNEL);
+    }
+
+    /**
+     * Creates a new simpleLed component with a custom BCM pin.
+     *
+     * @param pi4j    Pi4J context
```
[the remark this answers is not in the document]
Comment (Mikscho) [reply src/main/java/com/pi4j/example/components/LEDMatrix.java]: done

**Pull request merged**